src/server/pps/server/api_server.go

```diff
@@ -1067,6 +1069,229 @@ func (a *apiServer) DeleteAll(ctx context.Context, request *types.Empty) (respon
+				return addActiveTree(resp.Object)
+			})
+		}
+	}
+	if err := eg.Wait(); err != nil {
+		return nil, err
+	}
+
+	// Iterate through all objects.  If they are not active, delete them.
+	objects, err := objClient.ListObjects(ctx, &pfs.ListObjectsRequest{})
+	if err != nil {
+		return nil, err
+	}
+
+	var objectsToDelete []*pfs.Object
+	deleteObjectsIfMoreThan := func(n int) error {
+		if len(objectsToDelete) > n {
+			if _, err := objClient.DeleteObjects(ctx, &pfs.DeleteObjectsRequest{
+				Objects: objectsToDelete,
+			}); err != nil {
+				return fmt.Errorf("error deleting objects: %v", err)
+			}
+			objectsToDelete = []*pfs.Object{}
+		}
+		return nil
+	}
+	for object, err := objects.Recv(); err != io.EOF; object, err = objects.Recv() {
+		if err != nil {
+			return nil, fmt.Errorf("error receiving objects from ListObjects: %v", err)
+		}
+		if !activeObjects[object.Hash] {
+			objectsToDelete = append(objectsToDelete, object)
+		}
+		// Delete objects in batches
+		if err := deleteObjectsIfMoreThan(100); err != nil {
+			return nil, err
+		}
+	}
+	if err := deleteObjectsIfMoreThan(0); err != nil {
+		return nil, err
+	}
+
+	// Iterate through all tags.  If they are not active, delete them
+	tags, err := objClient.ListTags(ctx, &pfs.ListTagsRequest{})
+	if err != nil {
+		return nil, err
+	}
+	var tagsToDelete []string
+	deleteTagsIfMoreThan := func(n int) error {
+		if len(tagsToDelete) > n {
+			if _, err := objClient.DeleteTags(ctx, &pfs.DeleteTagsRequest{
+				Tags: tagsToDelete,
+			}); err != nil {
+				return fmt.Errorf("error deleting tags: %v", err)
+			}
+			tagsToDelete = []string{}
+		}
+		return nil
+	}
+	for resp, err := tags.Recv(); err != io.EOF; resp, err = tags.Recv() {
+		if err != nil {
+			return nil, fmt.Errorf("error receiving tags from ListTags: %v", err)
+		}
+		if !activeTags[resp.Tag] {
+			tagsToDelete = append(tagsToDelete, resp.Tag)
+		}
+		if err := deleteTagsIfMoreThan(100); err != nil {
+			return nil, err
+		}
+	}
+	if err := deleteTagsIfMoreThan(0); err != nil {
+		return nil, err
+	}
+
+	if err := a.incrementGCGeneration(ctx); err != nil {
+		return nil, err
+	}
+
+	return &pps.GCResponse{}, nil
+}
+
+// incrementGCGeneration increments the GC generation number in etcd
+func (a *apiServer) incrementGCGeneration(ctx context.Context) error {
+	resp, err := a.etcdClient.Get(ctx, client.GCGenerationKey)
+	if err != nil {
+		return err
+	}
+
+	if resp.Count == 0 {
+		// If the generation number does not exist, create it.
+		// It's important that the new generation is 1, as the first
+		// generation is assumed to be 0.
+		if _, err := a.etcdClient.Put(ctx, client.GCGenerationKey, "1"); err != nil {
+			return err
+		}
+	} else {
+		oldGen, err := strconv.Atoi(string(resp.Kvs[0].Value))
+		if err != nil {
+			return err
+		}
+		newGen := oldGen + 1
+		if _, err := a.etcdClient.Put(ctx, client.GCGenerationKey, strconv.Itoa(newGen)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
`this` meaning the `incrementGCGeneration` function?  It's used in the `GC` function above.

If you are talking about the generation number itself: it's used in `obj_block_api_server.go`.